lib/diagnostics_channel.js

```diff
@@ -105,11 +105,20 @@ class ActiveChannel {
     const index = ArrayPrototypeIndexOf(this._subscribers, subscription);
     if (index === -1) return false;
 
-    ArrayPrototypeSplice(this._subscribers, index, 1);
+    // Remove the subscriber but keep the array element
+    this._subscribers[index] = undefined;
 
     channels.decRef(this.name);
     maybeMarkInactive(this);
 
+    // Remove the element from the array once we finish any publish() loops.
+    process.nextTick(() => {
+      if (this._subscribers) {
+        this._subscribers = this._subscribers.filter((v) => v !== undefined);
```

Comment:
Primordials?

Reply:
I changed the code and it's now irrelevant.